src/Components/Web/src/FileDownloader.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Provides an abstraction for downloading a file from a Blazor app.
+/// </summary>
+public sealed class FileDownloader : IFileDownloader
+{
+    private IJSRuntime JS { get; set; } = default!;
+
+    /// <summary>
+    /// Initializes the <see cref="FileDownloader"/>.
+    /// </summary>
+    /// <param name="JSRuntime">The <see cref="IJSRuntime"/> to use for interoperability.</param>
+    public FileDownloader(IJSRuntime JSRuntime)
+    {
+        JS = JSRuntime;
+    }
+
+    /// <summary>
+    /// Takes in a byte[] representing file data and converts it into a MemoryStream.
+    /// </summary>
+    /// <param name="fileName">A <see cref="string"/> that contains the specified file name.</param>
+    /// <param name="data"> The <see cref="byte"/>[] data that will be converted into a <see cref="MemoryStream"/>.</param>
+    public Task SaveAs(string fileName, byte[] data)
```

Comment:
```suggestion
    /// <inheritdoc />
    public Task SaveAs(string fileName, byte[] data)
```

(same thing below as well)